weatherman.py

```diff
@@ -0,0 +1,201 @@
+import argparse
+import calendar
+import csv
+import glob
+import os
+from colorama import Fore, Style
+
+
+def convert_month(weather_month_input):
+    try:
+        month_number = int(weather_month_input)
+        if 1 <= month_number <= 12:
+            return calendar.month_abbr[month_number]
+    except ValueError:
+        raise ValueError("Invalid input: Please provide a valid month number between 1 and 12")
+
+
+def generate_weather_file_names(base_directory, weather_year, weather_month=None):
+    pattern = f"*{weather_year}*"
+    if weather_month:
+        pattern = f"*{weather_year}*{weather_month}*"
+
+    files_to_process = glob.glob(os.path.join(base_directory, pattern))
+    return files_to_process
+
+
+def read_weather_files(weather_year, weather_month):
```

Comment:
This method will only receive the weather file names

Reply:
yes this will receive weather files according to users demand and send it to read weather files for further calculations i used it to avoid reading data of all files instead it will allow us to only read data of files user asked